frontend/catalyst/compiler.py

```diff
@@ -339,7 +339,13 @@ def run_from_ir(
                func_name (str) Inferred name of the main function
                ret_type_name (str) Inferred main function result type name
         """
-        pipelines = pipelines if pipelines is not None else DEFAULT_PIPELINES
+        pipelines = (
+            pipelines
+            if pipelines is not None
+            else (
+                self.options.pipelines if self.options.pipelines is not None else DEFAULT_PIPELINES
+            )
+        )
```

Comment:
Users want to specify the pipelines via the parameters, but we also use an utility pipeline consisting of a single canonicalize call. So I thought it is better to handle the logic closer to the point where we use it. Do you think we should resolve pipelines outside this function?

Reply:
It is hard to follow given all the variables and fields named pipelines here.

I normally just prefer to have:

```python
if param is None:
     param = some_default
```

Why not reuse the single canonicalize call in `self.options.pipelines = canonicalize`? Or better, create a temporary CompilerOptions just for that canonicalize call.

Ideally this call is just like a library and different callers should not have to have different ways of having the same functionality.

Reply:
> if param is None:
>     param = some_default

Got it. I think I like oneliners because  typecheckers I used in other projects tend to issue better error messages on type mismatches. This is because Python requires programmers to always specify `else` branch in the `.. if .. else ..` expressions. For Catalyst this argument does not work. 

I have fixed this one by adding `pipelines` into the `CompilerOptions`. Should look better now, please check!

Reply:
Thank you.